src/include/duckdb/storage/compression/alp/alp_analyze.hpp

```diff
@@ -0,0 +1,190 @@
+}
+
+/*
+ * ALP Analyze step only pushes the needed samples to estimate the compression size in the finalize step
+ */
+template <class T>
+bool AlpAnalyze(AnalyzeState &state, Vector &input, idx_t count) {
+	auto &analyze_state = (AlpAnalyzeState<T> &)state;
+	UnifiedVectorFormat vdata;
+	input.ToUnifiedFormat(count, vdata);
+	auto data = UnifiedVectorFormat::GetData<T>(vdata);
+
+	//! We sample equidistant vectors; to do this we skip a fixed values of vectors
+	bool must_select_rowgroup_samples = (analyze_state.vectors_count % AlpConstants::RG_SAMPLES_DUCKDB_JUMP) == 0;
+	analyze_state.vectors_count += 1;
+	analyze_state.total_values_count += count;
+
+	//! If we are not in the correct jump, we do not take sample from this vector
+	if (!must_select_rowgroup_samples) {
+		return true;
+	}
+
+	//! We do not take samples of non-complete duckdb vectors (usually the last one)
+	//! Except in the case of too little data
+	if (count < AlpConstants::SAMPLES_PER_VECTOR && analyze_state.vectors_sampled_count != 0) {
+		return true;
+	}
+
+	uint32_t n_lookup_values = MinValue(count, (idx_t)AlpConstants::ALP_VECTOR_SIZE);
+	//! We sample equidistant values within a vector; to do this we jump a fixed number of values
+	uint32_t n_sampled_increments =
+	    MaxValue(1, (int32_t)std::ceil((double)n_lookup_values / AlpConstants::SAMPLES_PER_VECTOR));
+	uint32_t n_sampled_values = std::ceil((double)n_lookup_values / n_sampled_increments);
+
+	T a_non_null_value = 0;
+	idx_t nulls_idx = 0;
+
+	vector<uint16_t> current_vector_null_positions(n_lookup_values, 0);
+	vector<T> current_vector_values(n_lookup_values, 0);
+	vector<T> current_vector_sample(n_sampled_values, 0);
+
+	// Storing the entire sampled vector
+	//! We need to store the entire sampled vector to perform the 'analyze' compression in it
+	for (idx_t i = 0; i < n_lookup_values; i++) {
+		auto idx = vdata.sel->get_index(i);
+		T value = data[idx];
+		//! We resolve null values with a predicated comparison
+		bool is_null = !vdata.validity.RowIsValid(idx);
+		current_vector_null_positions[nulls_idx] = i;
+		nulls_idx += is_null;
+		current_vector_values[i] = value;
+	}
+
+	// Finding the first non-null value
+	idx_t tmp_null_idx = 0;
+	for (idx_t i = 0; i < n_lookup_values; i++) {
+		if (i != current_vector_null_positions[tmp_null_idx]) {
+			a_non_null_value = current_vector_values[i];
+			break;
+		}
+		tmp_null_idx += 1;
+	}
+
+	// Replacing that first non-null value on the vector
+	for (idx_t i = 0; i < nulls_idx; i++) {
+		uint16_t null_value_pos = current_vector_null_positions[i];
+		current_vector_values[null_value_pos] = a_non_null_value;
+	}
+
+	// Storing the sample of that vector
+	idx_t sample_idx = 0;
+	for (idx_t i = 0; i < n_lookup_values; i += n_sampled_increments) {
+		current_vector_sample[sample_idx] = current_vector_values[i];
+		sample_idx++;
+	}
+	D_ASSERT(sample_idx == n_sampled_values);
+
+	analyze_state.complete_vectors_sampled.push_back(current_vector_values);
```

Comment:
I have double checked and what you say is true. It will create a copy of the vector. By using `std::move` this won’t happen (since C++ 11).
We added the `std::move()`. The compiler did not complained at all.